### PR TITLE
Make extract_return_annotation compatible with 3.9 NamedTuple

### PR DESCRIPTION
**Pull request by fediazgon**
# TL;DR
Make `extract_return_annotation` compatible with Python 3.9 `NamedTuple`.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
`_field_type` attribute was removed in Python 3.9 (see [typing.NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple)). Instead, check that the type has `tuple` as a base class, and that it has the `_field` attribute (part of [namedtuple API](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._fields)). Then return `__annotations__` which contains the same information as `_field_types`. 

## Tracking Issue
https://github.com/flyteorg/flyte/issues/914

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]
Is this the only problem with using Python 3.9 you guys have found?
